### PR TITLE
Removing .NET 7 from tests, removing outdated tests and increasing timeout-minutes

### PR DESCRIPTION
**Pull request by nicrowe00**
Cleaning up tests by removing references to .NET 7 and removing tests
made for older, now unsupported versions of .NET. Also increasing
the timeout limit for install .NET and test dependencies due to frequent
timeouts on Fedora.

### CODE REVIEW

File: managed-symbols-available/test.json [hunk 1]
```diff
@@ -8,8 +8,7 @@
   "cleanup": true,
   "skipWhen": [
     "vmr-ci,version=6", // unpacking symbols is a hassle
-    "vmr-ci,version=7",
-    "non-packaged",
+    "non-packaged"
```
Comment (omajid) [inline managed-symbols-available/test.json:11]: FYI: we often leave the trailing comma - even though that is not allowed (along with the `//` comments) by strict JSON implementations - to make it easier to add items to the list without more diff noise.

File: debugging-via-dotnet-dump/test.sh [hunk 2]
```diff
@@ -319,18 +319,6 @@ grep 'Attributes: *PEFile' dump.out
...
-# dump-analyze 'name2ee *!System.String' > dump.out
-# cat dump.out
-# grep 'MethodTable:' dump.out | awk '{print $2}'
-# string_method_table=$(grep 'MethodTable:' dump.out | awk '{print $2}')
-# dump-analyze "dumpmt ${string_method_table}" > dump.out
-# cat dump.out
-# grep -E '^Name:[ \n\t]+System.String' dump.out
-# grep -E "^File:[ \n\t]+${framework_dir}" dump.out
-
-
```
Comment (omajid) [inline debugging-via-dotnet-dump/test.sh:333]: Any chance we can review and then enable this for 8.0 and later?

File: managed-symbols-available/test.json (hunk 1 above)
Comment (nicrowe00) [reply managed-symbols-available/test.json]: Noted, thanks!

File: debugging-via-dotnet-dump/test.sh (hunk 2 above)
Comment (nicrowe00) [reply debugging-via-dotnet-dump/test.sh:333]: I enabled it and tested for .NET 6 and 8, seems to be working correctly.

Comment (omajid) [reply debugging-via-dotnet-dump/test.sh:333]: Okay, this seems to be failing. My slight preference is to leave it commented out (like it was before this PR) so we can investigate it separately in the future. But I am okay if you want to remove it too.

**Comment by omajid** [comment]
I opened https://github.com/redhat-developer/dotnet-regular-tests/pull/351 to remove the failing centos stream 8 legs. CentOS Stream 8 is EOL: https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/

**Review by omajid — APPROVED** [review]
Thank you!!

**Pull request merged**